code/GatewayFieldsFactory.php

```diff
@@ -320,47 +340,118 @@ protected function cullForGateway(&$fields, $defaults = array())
      *
      * @return array
      */
-    public function normalizeFormData(array $data) {
-
-        $renameMap = \Config::inst()->get('GatewayFieldsFactory', 'rename');
-        
-        if (!$renameMap || empty($renameMap)) {
+    public function normalizeFormData(array $data)
+    {
+        if (empty($data)) {
             return $data;
         }
 
-        $hasPrefix = false;
+        $renameMap = array_flip(static::$renamemap);
 
-        if (array_key_exists('prefix', $renameMap)) {
-            $hasPrefix = true;
+        foreach ($renameMap as $customName => $defaultName) {
+            if (array_key_exists($customName, $data)) {
+                $value = $data[$customName];
+                unset($data[$customName]);
+                $data[$defaultName] = $value;
+            }
         }
 
-        // Fix for: array_flip(): Can only flip STRING and INTEGER values!
-        // This would occur if you have defined Gateways in rename.yml
-        foreach($globalMap = $renameMap as $key => $value) {
-            if (is_array($value)) {
-                unset($globalMap[$key]);
-            }
+        return $data;
+    }
+
+    /**
+     * Fetches the prefix (if defined) from the global section of the rename map
+     * @return mixed
+     */
+    public function getGlobalPrefix()
+    {
+        return $this->getGlobalFieldName('prefix');
+    }
+
+    /**
+     * Fetches custom name from the rename map, or returns false
+     *
+     * @param string $defaultName The original name of the field
+     *
+     * @return bool|string Returns false if no custom name has been defined, otherwise returns the custom name
+     */
+    public function getGlobalFieldName($defaultName)
```

Comment:
This exists purely to reduce the large cyclomatic entry points within `getFieldName`, and is used to build the map initially.

The only `$this->renamemap` accessor should be `getFieldName()` as you have mentioned below

Maybe `private` scope would be wiser